### ISSUE TITLE
[Accessibility Audit] Better error messages for forms

### ISSUE THREAD

**Issue by SaptakS** [freedomofpress/pressfreedomtracker.us#2007]
Currently we do have the default browser error message for certain form requirements showing up, but they are not very accessible and we should have our own error message implementation for the forms.

<img width="660" height="391" alt="Image" src="https://github.com/user-attachments/assets/5e4a7f5f-7219-4020-9c47-768975a8ffe8" />
